repos/spack_repo/builtin/packages/charmpp/package.py

```diff
@@ -376,6 +376,8 @@ def install(self, spec, prefix):
         # been implemented in v8.0.0
         if self.spec.satisfies("@8.0.0: %aocc"):
             options.append("--disable-fortran")
+        if self.spec.satisfies("@8.0.0: ^[virtuals=fortran] intel-oneapi-compilers"):
```

Comment:
Isn't it a bit unexpected to concretize with a fortran compiler, and then disable fortran in the `install` method? Can we use directives, and enforce these constraints at concretization time?

What I'm thinking is:
```python
variant("fortran", default=False, description="Enable Fortran support")
depends_on("fortran", type="build", when="+fortran")
conflicts("%fortran=intel-oneapi-compilers", when="+fortran")
```

Reply:
@alalazo , yes a bit unexpected 😄  It would be more logical to disable dependency if we disable fortran.
`--disable-fortran` appeared only in version 8.0.0. If “when” can be used to define the variant, it can be done.

@matthiasdiener , What are your thoughts on that? If you’re okay, I can modify the recipe accordingly.

Reply:
> If “when” can be used to define the variant, it can be done.

Yes, you can:
```python
variant("fortran", default=False, when="@8:", description="Enable Fortran support")
```